x-pack/plugins/apm/public/components/shared/TraceLink.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+// @ts-ignore
+import { legacyEncodeURIComponent, RelativeLink } from '../../utils/url';
+
+interface Props {
+  name: string;
+  serviceName?: string;
+  transactionType?: string;
+  transactionId?: string;
+  traceId?: string;
+}
+
+export const TraceLink: React.SFC<Props> = ({
+  name,
+  serviceName,
+  transactionType,
+  transactionId,
+  traceId,
+  children
+}) => {
+  if (!serviceName || !transactionType) {
+    return <React.Fragment>{children}</React.Fragment>;
+  }
+
+  return (
+    <RelativeLink
```

Comment:
I was hoping we could deprecate `RelativeLink` for `KibanaLink` (which might need some re-work).

Reply:
Yeah right now we can't add query params to KibanaLink ... I'll see what we have time to do but I am worried about dragging this out too much so it may stay for now and get refactored later.